rest-api/src/main/java/com/cloudbees/workflow/flownode/FlowNodeUtil.java

```diff
@@ -44,33 +61,88 @@
  */
 public class FlowNodeUtil {
 
+    protected static final Cache<String,List<FlowNode>> executionCache = CacheBuilder.newBuilder().maximumSize(100).build();
```

Comment:
:bug: a `FlowNode` will in general hold a strong reference to a `WorkflowRun` and thus also `WorkflowJob`, which is not something you want in a `static` cache.